### PR TITLE
Yan-734 Optional Port Type

### PR DESCRIPTION
**Pull request by calgray**
This changeset updates the port EAGLE palette doxygen string format to:
```
*  \param[Direction] port/Name
and
*  \param[Direction] port/Name/Data Type
```
 as described in https://icrar.atlassian.net/wiki/spaces/DALIUGE/pages/506625791/Component+Doxygen+Comments

### CODE REVIEW

File: tools/xml2palette/xml2palette.py [hunk 1]
```diff
@@ -310,11 +324,11 @@ def process_compounddef(compounddef):
                 direction = pichild[0].attrib.get("direction", "").strip()
             elif pichild.tag == "parameterdescription":
                 if key == "gitrepo":
-                    if pichild[0][0].text is None:
+                    if pichild[0].text is None:
                         print("No gitrepo text")
                         value = ""
                     else:
-                        value = pichild[0][0].text.strip()
+                        value = pichild[0].text.strip()
```
Comment (calgray) [inline tools/xml2palette/xml2palette.py:331]: I assumed the gitrepo doc for example expands from `\param gitrepo $(GIT_REPO)` to `\param gitrepo https://github.com/ICRAR/EAGLE`

File: tools/xml2palette/xml2palette.py [hunk 2]
```diff
@@ -247,9 +253,17 @@ def process_compounddef(compounddef):
...
+                    "key":"text",
+                    "direction":None,
+                    "value":""
+                })
             else:
-                result.append({"key":"text", "direction":None, "value":briefdescription[0].text.strip()})
+                result.append({
+                    "key":"text",
+                    "direction":None,
+                    "value":briefdescription[0].text.strip(" .")
```
Comment (calgray) [inline tools/xml2palette/xml2palette.py:265]: here I added stripping of '.' which doxygen automatically adds for the \brief tag

Comment (james-strauss-uwa) [reply tools/xml2palette/xml2palette.py:265]: Looks good

File: tools/xml2palette/xml2palette.py (hunk 1 above)
Comment (james-strauss-uwa) [reply tools/xml2palette/xml2palette.py:331]: It's hard to tell what is correct here, I don't remember why the extra [0] is there.

Let's go with your changes though, and we'll revert if something goes wrong.

**Review by james-strauss-uwa — APPROVED** [review]
Great, thanks, merge it

**Pull request merged**